tests/tests/kernel/datatypes/ezimage/ezimagealiashandler_regression.php

```diff
@@ -88,6 +82,83 @@ public function testIssue15155()
         array_map( 'unlink', $toDelete );
         $image->purge();
     }
+
+    /**
+     * @return array
+     */
+    public function providerForTestImageNameObjectFalse()
+    {
+        return array(
+            array(
+                str_repeat( 'a', 180 ) . "This is a Very Long Name isn't it?",
+                str_repeat( 'a', 180 ) . "This-is-a-Very-Long-1"
+            ),
+            array(
+                str_repeat( 'a', 180 ) . "私は簡単にパブリッシュの記事で使用することができるようなも",
+                str_repeat( 'a', 180 ) . "私は簡単にパ1"
+            ),
+            array(
+                str_repeat( 'a', 180 ) . "私aはb簡c単dにeパfブgリhッシュの記事で使用することがで",
+                str_repeat( 'a', 180 ) . "私aはb簡c単dにe1"
+            )
+        );
+    }
+
+    public function providerForTestImageNameByNodeObjectFalse()
+    {
+        return array(
+            array(
+                str_repeat( 'a', 180 ) . "This is a Very Long Name isn't it?",
+                str_repeat( 'a', 180 ) . "This-is-a-Very-Long-"
+            ),
+            array(
+                str_repeat( 'a', 180 ) . "私は簡単にパブリッシュの記事で使用することができるようなも",
+                str_repeat( 'a', 180 ) . "私は簡単にパ"
+            ),
+            array(
+                str_repeat( 'a', 180 ) . "私aはb簡c単dにeパfブgリhッシュの記事で使用することができるようなも",
+                str_repeat( 'a', 180 ) . "私aはb簡c単dにe"
+            )
+        );
+    }
+
+    /**
+     * @dataProvider providerForTestImageNameObjectFalse
+     */
+    public function testImageNameObjectFalse( $longName, $expects )
+    {
+        ezpINIHelper::setINISetting( 'site.ini', 'URLTranslator', 'TransformationGroup', 'urlalias_iri' );
+
+        $handler = new eZImageAliasHandler( null );
+        $language = "fre-FR";
+        $contentVersionMock = $this->getMockBuilder( 'eZContentObjectVersion' )->disableOriginalConstructor()->getMock();
+        $contentVersionMock->expects( $this->any() )->method( 'versionName' )->willReturn( $longName );
```

Comment:
use `will()` instead of `willReturn()` (not compatible with required PHPUnit version in legacy)